### PR TITLE
8344540: Remove superseded wildcard description from java manpage

### PR DESCRIPTION
**Pull request by sormuras**
Please review this follow-up PR to improve `java`'s manpage section about wildcards in argument files: the confusing sentence is now removed.

This is a cleanup of commit https://github.com/openjdk/jdk/commit/5cb0d438231383d491b2fcca455d09af7f2ee016

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344540](https://bugs.openjdk.org/browse/JDK-8344540): Remove superseded wildcard description from java manpage (**Enhancement** - P5)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22236/head:pull/22236` \
`$ git checkout pull/22236`

Update a local copy of the PR: \
`$ git checkout pull/22236` \
`$ git pull https://git.openjdk.org/jdk.git pull/22236/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22236`

View PR using the GUI difftool: \
`$ git pr show -t 22236`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22236.diff">https://git.openjdk.org/jdk/pull/22236.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22236#issuecomment-2485534860)
</details>

### CODE REVIEW

**Comment by jaikiran** [comment]
Hello Christian, I think this change in the description of the syntax for arg files is accurate and removes any confusion. 

While at it, I think we should also do a small update to the following line in the section "Using the JDK\_JAVA\_OPTIONS Launcher Environment Variable":

> However, as in `@`-files, use of a wildcard is not supported.

Based on what I see in the launcher code which handles the `JDK_JAVA_OPTIONS` environment variable, I think we should just change it to:

> The wildcard literal `*` isn't expanded and is passed as-is to the `java` launcher.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Can I also suggest some grammatical tweaks to the arg file section:
```
    The launcher doesn't expand wildcards that are present within an argument
    file. That means, an asterisk  `*` is passed on as-is to the starting VM.
```
There is an extra space after "asterisk". No comma needed after "means" and the asterisk character should be in parentheses e.g.
```
    file. That means an asterisk (`*`) is passed on as-is to the starting VM.
```

```
    For example `*.java` stays `*.java` and is not expanded to `Foo.java`,
    `Bar.java`, etc. like on some command line shell.
```
Suggestion
```
    For example `*.java` stays `*.java` and is not expanded to `Foo.java
    Bar.java ...`,  as would happen with some command line shells.
```

**Comment by sormuras** [comment]
@jaikiran and @dholmes-ora - I updated the wording according to your suggestions.

File: src/java.base/share/man/java.md
Comment (dholmes-ora) [inline src/java.base/share/man/java.md:3077]:
```suggestion
    file. That means an asterisk (`*`) is passed on as-is to the starting VM.
```

File: src/java.base/share/man/java.md
Comment (dholmes-ora) [inline src/java.base/share/man/java.md:3079]: The list was intended to be contiguous - no commas: \'Foo.java Bar.java ...\' as that is how it would be expanded by the shell.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Couple more nits sorry.

File: src/java.base/share/man/java.md
Comment (sormuras) [reply src/java.base/share/man/java.md]: Addressed this and the other one in https://github.com/openjdk/jdk/pull/22236/commits/de0ac7c9676e405158c375f469ce1157bdc6f033

File: src/java.base/share/man/java.md
Comment (sormuras) [reply src/java.base/share/man/java.md]: Addressed this and the other one in https://github.com/openjdk/jdk/pull/22236/commits/de0ac7c9676e405158c375f469ce1157bdc6f033

**Review by jaikiran — APPROVED** [review]
Thank you for the updates Christian. This looks good to me. Please wait for David's approval before integrating.

**Review by dholmes-ora — APPROVED** [review]
Ship it! Thanks!

**Pull request closed without merging**
